cmd/snap/cmd_run.go

```diff
@@ -464,10 +475,20 @@ func (x *cmdRun) snapRunApp(snapApp string, args []string) error {
 		return fmt.Errorf(i18n.G("cannot find app %q in %q"), appName, snapName)
 	}
 
+	if !app.IsService() {
+		if err := maybeWaitWhileInhibited(snapName); err != nil {
+			return err
+		}
+	}
+
 	return x.runSnapConfine(info, app.SecurityTag(), snapApp, "", args)
 }
 
 func (x *cmdRun) snapRunHook(snapName string) error {
+	if err := maybeWaitWhileInhibited(snapName); err != nil {
```

Comment:
as is this would stop us from running the gate-auto-refresh hook,  we need to think a bit, also we probably don't want the   the graphical UI for hooks??

Reply:
You're right, removed (no waiting for hooks).